src/server/game/Entities/Player/Player.cpp

```diff
@@ -16460,7 +16460,7 @@ bool Player::HasQuestForItem(uint32 itemid) const
             // This part for ReqItem drop
             for (uint8 j = 0; j < QUEST_ITEM_OBJECTIVES_COUNT; ++j)
             {
-                if (itemid == qinfo->RequiredItemId[j] && q_status.ItemCount[j] < qinfo->RequiredItemCount[j])
+                if ((itemid == qinfo->RequiredItemId[j] && q_status.ItemCount[j] < qinfo->RequiredItemCount[j]) || (turnIn && itemid == qinfo->RequiredItemId[j] && q_status.ItemCount[j] <= qinfo->RequiredItemCount[j]))
```

Comment:
The same applies here, duplicated `itemid == qinfo->RequiredItemId[j]` is not needed
`if (itemid == qinfo->RequiredItemId[j] && (q_status.ItemCount[j] < qinfo->RequiredItemCount[j] || (turnIn && q_status.ItemCount[j] <= qinfo->RequiredItemCount[j])))`
I would also want to know why are you using `<=` comparison during quest turn in, what does it change?